src/record.rs

```diff
@@ -99,13 +121,95 @@ impl<'a> Serializable<'a> for MultiPlexedRecord<'a> {
                 position,
                 payload,
             }),
+            RecordType::AppendRecords => Some(MultiPlexedRecord::AppendRecords {
+                queue,
+                position,
+                records: MultiRecord::new(payload)?,
+            }),
             RecordType::Truncate => Some(MultiPlexedRecord::Truncate { queue, position }),
             RecordType::Touch => Some(MultiPlexedRecord::Touch { queue, position }),
             RecordType::DeleteQueue => Some(MultiPlexedRecord::DeleteQueue { queue, position }),
         }
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct MultiRecord<'a> {
+    buffer: &'a [u8],
+    position: usize,
```

Comment:
naming that position will make the codebase very confusing.
 
```suggestion
    position: usize, 
```
```suggestion
    byte_offset: usize,
```